### PR TITLE
Issue #2477 - Update to Java 11

### PR DESCRIPTION
**Pull request by tbieste**
Issue #2477 - Update to Java 11

Signed-off-by: Troy Biesterfeld <tbieste@us.ibm.com>

### CODE REVIEW

File: fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/streams/Streamer.java [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 /**
  * Simple implementation taking a JDBC ResultSet and rendering it using
- * the Java 8 Streams API.
+ * the Java 11 Streams API.
```
Comment (lmsurpre) [inline fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/streams/Streamer.java:22]: maybe better to just take it out in this case
```suggestion
 * the Java Streams API.
```

File: docs/src/pages/guides/FHIRServerUsersGuide.md [hunk 2]
```diff
@@ -44,7 +44,7 @@ This FHIR server is intended to be a common component for providing FHIR capabil
 # 2 Installation
 
 ## 2.1 Installing a new server
-0.  Prereqs: The IBM FHIR Server requires Java 8 or higher and has been tested with OpenJDK 8, OpenJDK 11, and the IBM SDK, Java Technology Edition, Version 8. To install Java on your system, we recommend downloading and installing OpenJDK 8 from https://adoptopenjdk.net/.
+0.  Prereqs: The IBM FHIR Server requires Java 11 or higher, and has been tested with OpenJDK 11. To install Java on your system, we recommend downloading and installing OpenJDK 11 from https://adoptopenjdk.net/.
```
Comment (prb112) [inline docs/src/pages/guides/FHIRServerUsersGuide.md:47]:
```suggestion
0.  Prereqs: The IBM FHIR Server requires Java 11 and has been tested with OpenJDK 11. To install Java on your system, we recommend downloading and installing OpenJDK 11 from https://adoptopenjdk.net/.
```

File: fhir-parent/pom.xml [hunk 3]
```diff
@@ -563,8 +563,7 @@
             <dependency>
                 <groupId>com.github.ben-manes.caffeine</groupId>
                 <artifactId>caffeine</artifactId>
-                <!-- upgrade to 3.x after we discontinue Java 8 support -->
-                <version>2.9.0</version>
+                <version>3.0.5</version>
```
Comment (prb112) [inline fhir-parent/pom.xml:566]: Excellent

**Review by prb112 — APPROVED** [review]
1 comment in place.

Also, please check the README.md

**Comment by prb112** [comment]
> I'm hitting a bunch of "The package java.util is accessible from more than one module: , java.base" errors when I switch the compatibility level from Java 8 to Java 11. Trying to figure out how to resolve it.

I'd recommend doing a build of our code, and then scanning the Jars to see if one repackages java.util

File: fhir-tools/pom.xml [hunk 4]
```diff
@@ -17,7 +17,7 @@
         <jacoco.plugin.version>0.8.2</jacoco.plugin.version>
         <argLine>-Dfile.encoding=UTF-8</argLine>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <java.version>1.8</java.version>
+        <java.version>11</java.version>
         <maven.build.timestamp.format>yyyyMMdd-HHmm</maven.build.timestamp.format>
         <maven.deploy.skip>true</maven.deploy.skip>
```
Comment (prb112) [inline fhir-tools/pom.xml:22]: <source>8</source> update this in maven-javadoc-plugin

File: fhir-examples/pom.xml [hunk 5]
```diff
@@ -43,7 +43,7 @@
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <java.version>1.8</java.version>
+        <java.version>11</java.version>
         <maven.compiler.source>${java.version}</maven.compiler.source>
         <maven.compiler.target>${java.version}</maven.compiler.target>
```
Comment (prb112) [inline fhir-examples/pom.xml:48]: <source>8</source> update this in maven-javadoc-plugin

File: fhir-parent/pom.xml [hunk 6]
```diff
@@ -47,7 +47,7 @@
         <fhir-search.index>MINIMAL_JSON</fhir-search.index>
         <fhir-persistence-jdbc.index>MINIMAL_JSON</fhir-persistence-jdbc.index>
         <fhir-server-test.index>MINIMAL_JSON</fhir-server-test.index>
-        <java.version>1.8</java.version>
+        <java.version>11</java.version>
```
Comment (prb112) [inline fhir-parent/pom.xml:50]: <source>8</source> update this in maven-javadoc-plugin

Comment (tbieste) [reply fhir-parent/pom.xml:50]: Updated.

File: fhir-examples/pom.xml (hunk 5 above)
Comment (tbieste) [reply fhir-examples/pom.xml:48]: Updated.

File: fhir-tools/pom.xml (hunk 4 above)
Comment (tbieste) [reply fhir-tools/pom.xml:22]: Updated.

File: docs/src/pages/guides/FHIRServerUsersGuide.md (hunk 2 above)
Comment (tbieste) [reply docs/src/pages/guides/FHIRServerUsersGuide.md:47]: Updated.

File: fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/streams/Streamer.java (hunk 1 above)
Comment (tbieste) [reply fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/streams/Streamer.java:22]: Updated.

**Comment by tbieste** [comment]
Ah, here it is:  https://github.com/JanusGraph/janusgraph/issues/2900

**Comment by prb112** [comment]
> Ah, here it is: [JanusGraph/janusgraph#2900](https://github.com/JanusGraph/janusgraph/issues/2900)

Please wait... I have a PR that updates JanusGraph

**Review by prb112 — APPROVED** [review]
LGTM - great changes

File: docs/src/pages/guides/FHIRModelGuide.mdx [hunk 7]
```diff
@@ -6,7 +6,7 @@ date:  "2019-10-25 12:00:00 -0400"
 
 ## Overview
 
-The FHIR model component provides Java APIs for parsing, building, generating and validating FHIR resources. Java model classes that represent FHIR resources and data types are generated directly from the structure definitions distributed with the spec. All model objects are thread-safe and immutable. Each model class implements the Java builder pattern (Effective Java, Joshua Bloch) and the visitor pattern (GoF). The classes also implement Java equals, hashCode and toString methods. All date/time processing is done using the Java 8 time library.
+The FHIR model component provides Java APIs for parsing, building, generating and validating FHIR resources. Java model classes that represent FHIR resources and data types are generated directly from the structure definitions distributed with the spec. All model objects are thread-safe and immutable. Each model class implements the Java builder pattern (Effective Java, Joshua Bloch) and the visitor pattern (GoF). The classes also implement Java equals, hashCode and toString methods. All date/time processing is done using the Java 11 time library.
```
Comment (lmsurpre) [inline docs/src/pages/guides/FHIRModelGuide.mdx:9]: missed this one before, but probably should just be "Java time library" (similar to that "Java 8 streams" one)

File: docs/src/pages/guides/FHIRServerUsersGuide.md [hunk 8]
```diff
@@ -44,7 +44,7 @@ This FHIR server is intended to be a common component for providing FHIR capabil
 # 2 Installation
 
 ## 2.1 Installing a new server
-0.  Prereqs: The IBM FHIR Server requires Java 8 or higher and has been tested with OpenJDK 8, OpenJDK 11, and the IBM SDK, Java Technology Edition, Version 8. To install Java on your system, we recommend downloading and installing OpenJDK 8 from https://adoptopenjdk.net/.
+0.  Prereqs: The IBM FHIR Server requires Java 11 and has been tested with OpenJDK 11. To install Java on your system, we recommend downloading and installing OpenJDK 11 from https://adoptopenjdk.net/.
```
Comment (lmsurpre) [inline docs/src/pages/guides/FHIRServerUsersGuide.md:47]: doesn't have to be in this PR, but we should update this to point to adoptium
```suggestion
0.  Prereqs: The IBM FHIR Server requires Java 11 and has been tested with OpenJDK 11. To install Java on your system, we recommend downloading and installing OpenJDK 11 from https://adoptium.net/.
```

File: docs/src/pages/guides/FHIRModelGuide.mdx (hunk 7 above)
Comment (tbieste) [reply docs/src/pages/guides/FHIRModelGuide.mdx:9]: Updated.

**Pull request merged**